spec/policies/documentation_policy_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+describe DocumentationPolicy do
+  subject { described_class.new(user, record) }
+
+  let!(:parent_division) { create(:division) }
+  let!(:division) { create(:division, parent: parent_division) }
+  let!(:child_division) { create(:division, parent: division) }
+  let(:user) { create(:user, :member, division: division) }
+
+  context "documentation belongs to user's division" do
+    let(:record) { create(:documentation, division: division) }
+
+    it { should permit_all }
+  end
+
+  context "documentation belongs to ancestor of user's division" do
+    let(:record) { create(:documentation, division: parent_division) }
+
+    it { should permit_action(:show) }
+    it { should forbid_action(:edit) }
+  end
+
+  context "documentation belongs to descendant of user's division" do
+    let(:record) { create(:documentation, division: child_division) }
+
+    it { should permit_action(:show) }
+    it { should permit_action(:edit) }
+  end
+
+  context "documentation belongs to a division that is not a descendant or ancestor of any of the user's divisions " do
+    let(:other_division) { create(:division) }
+    let(:record) { create(:documentation, division: other_division) }
+    it { should forbid_action(:show) }
+    it { should forbid_action(:edit) }
+  end
```

Comment:
I started trying to write a spec for the case that documentation belongs to an ancestor of another division the user has access to (besides the `user.division`). Then I looked for examples of users having access to divisions other than the one they are assigned to, and couldn't find them. The DivisionPolicy `base_accessible_ids` method looks for any division where the user is a member or admin. Is there an example of a user having a role on a division that they aren't assigned to? Does that happen in practice?  This is what I started writing: ```context "documentation belongs to an ancestor of another division the user has access to" do
    let(:other_parent_division) { create(:division) }
    let(:other_division) { create(:division, parent: other_parent_division) }
    let(:record) { create(:documentation, division: other_parent_division) }
    # give user a role on `other_division` while keeping user.division `division`

    it { should permit_action(:show) }
    it { should permit_action(:edit) }
  end```

Reply:
I'm generally a little confused about how :admin and :member should play into edit permissions here, but maybe that's not something we need to tackle in this issue. If we do, it will involve revisiting the use of `base_accessible_id`, which doesn't distinguish between the two. We could have edit permissions on documentation use division_admin? like many other resources, instead, but I am assuming there was some good reason for members to be able to edit documentations/documentations to have their permissions own logic.

Reply:
I think we should copy the show logic from the question policy and just leave edit at the default for the application policy for now. This is not a high stakes permissions issue, but I think that brendan is looking for behavior similar to the question policy here and it might be easier to just reuse it rather than continuing to go to trouble trying to sort out the right logic. I don't think the question policy's logic is written correctly for what it's trying to do, but it does seem to work okay for now, and there's no way the client wants us spending much more extra time on this.

Don't bother with the test case for giving a user a role outside of their tree, the basics are good enough here, lets not overthink it.

Reply:
ok, great - the question logic lines up with the specs I had in there, and I like removing the more complex code from the documentation policy a lot.